### PR TITLE
feat(dbt-fal): optimize bigquery read and write

### PR DESCRIPTION
**Pull request by chamini2**
## Description

Use solution heavily based on fal lib.py

### CODE REVIEW

File: adapter/src/dbt/adapters/fal_experimental/support/bigquery.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                # TODO: offer as a dbt.config parameter?
+                # bigquery.SchemaField.from_api_repr(field)
+                # for field in (fields_schema or [])
+            ],
+            source_format="PARQUET",
+            write_disposition=WriteDisposition.WRITE_TRUNCATE,
+        )
+
+        with connection_manager.exception_handler("START JOB"):
+            job = client.load_table_from_dataframe(
```
Comment (mederka) [inline adapter/src/dbt/adapters/fal_experimental/support/bigquery.py:61]: ATL: so internally this is creating parquet files?

File: adapter/src/dbt/adapters/fal_experimental/support/bigquery.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                data, table_ref, job_config=job_config
+            )
+
+        timeout = connection_manager.get_job_execution_timeout_seconds(conn) or 300
+
+        with connection_manager.exception_handler("LOAD TABLE"):
```
Comment (mederka) [inline adapter/src/dbt/adapters/fal_experimental/support/bigquery.py:67]: Should we log something in debug mode here?

**Review by mederka — APPROVED** [review]
Great job, @chamini2! 🙏

File: adapter/src/dbt/adapters/fal_experimental/support/bigquery.py (hunk 1 above)
Comment (chamini2) [reply adapter/src/dbt/adapters/fal_experimental/support/bigquery.py:61]: It's downloading the data from the BQ client directly as a dataframe

File: adapter/src/dbt/adapters/fal_experimental/support/bigquery.py (hunk 2 above)
Comment (chamini2) [reply adapter/src/dbt/adapters/fal_experimental/support/bigquery.py:67]: I think it's fine

**Pull request merged**